file/cache.lisp

```diff
@@ -41,27 +52,52 @@
 (defmethod output-stream-p ((stream file-cache-stream))
   (member (direction stream) '(:output :io)))
 
+(defgeneric fs-read-block (stream block-n)
+  (:documentation "Read block-n from file"))
+
+(defgeneric fs-write-block (stream)
```

Comment:
This should take the block buffer & number as arguments instead of requiring that the method call dirty-block & dirty-block-n. Those look like they're implementation details of the cache and might change in the future when it
is able to manage multiple dirty blocks.